idc/views.py

```diff
@@ -264,11 +284,29 @@ def explore_data_page(request):
         collapse_on = req.get('collapse_on', 'SeriesInstanceUID')
         is_json = (req.get('is_json', "False").lower() == "true")
         uniques = json.loads(req.get('uniques', '[]'))
+        totals = json.loads(req.get('totals', '[]'))
+
         record_limit = int(req.get('record_limit', '2000'))
         offset = int(req.get('offset', '0'))
-
+        #sort_on = req.get('sort_on', collapse_on+' asc')
+        cohort_id = req.get('cohort_id','-1')
+
+        cohort_filters={}
+        if (int(cohort_id)>-1):
+            cohort = Cohort.objects.get(id=cohort_id, active=True)
+            cohort.perm = cohort.get_perm(request)
+            if cohort.perm:
+                wcohort = True
+                cohort_filters_dict = cohort.get_filters_as_dict()
+                cohort_filters_list = cohort_filters_dict[0]['filters']
+                for cohort in cohort_filters_list:
+                    cohort_filters[cohort['name']] = cohort['values']
+
+
+        if wcohort and is_json:
+            filters = cohort_filters
         context = build_explorer_context(is_dicofdic, source, versions, filters, fields, order_docs, counts_only,
-                                         with_related, with_derived, collapse_on, is_json, uniques=uniques)
+                                         with_related, with_derived, collapse_on, is_json, uniques=uniques, totals=totals)
```

Comment:
There's no need to ask for totals in the explorer context call; they're saved in the cohort. You can do it above (where indicated).

Reply:
But right now I am reporting the totals  by applying the old filter on the new data version BEFORE the new cohort is actually created. Example :clicking the 'Compare' button on cohort 4 on the /cohorts/ page opens this dialog which shows totals for the actual cohort 4 and those of the cohort that would be created by applying the filter to the current data version. But that new cohort doesn't exist yet. Clicking the button 'Load New Version' just goes to the explorer page with that filter preselected. The user still has to save the cohort. Btw ignore those zeros in screenshot - they're due to my issue running the stats script.
![Screen Shot 2021-04-30 at 5 03 01 PM](https://user-images.githubusercontent.com/58443591/116764434-efb9ee00-a9d5-11eb-9229-71ada0a607b6.png)

Reply:
For that you definitely don't want to run the entire context builder, as it results in a large amount of unneeded data being pulled (records, etc.) explicitly for displaying the main page.

All you'll want to do there is use _get_cohort_stats(), sending in 0 for a cohort ID and then the cohort filters and the data sources for the new version. So something like:

```
from cohorts import _get_cohort_stats

new_version_stats = _get_cohort_stats(
   0,
   old_cohort.cohort.get_filters_as_dict()[0]['filters'], 
   ImagingDataCommons.objects.get(active=True).get_data_sources(active=True, source_type=DataSource.SOLR, aggregate_level="StudyInstanceUID")
)
```

No need for the explorer view--it's not intended to be run at any time except to actually load the full explorer page.